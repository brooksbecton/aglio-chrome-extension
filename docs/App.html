<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: App</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: App</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>App<span class="signature">()</span><span class="type-signature"></span></h2>
        
            <div class="class-description">Wrapper for application</div>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    
    <h2>Constructor</h2>
    

    
    <h4 class="name" id="App"><span class="type-signature"></span>new App<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Creates an instance of ScraperFactory and attaches it to component
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line10">line 10</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    

    

    

    

    

    

    
        <h3 class="subsection-title">Members</h3>

        
            
<h4 class="name" id="saveRecipe"><span class="type-signature">(async) </span>saveRecipe<span class="type-signature"></span></h4>




<div class="description">
    Save the component's state to the database
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line38">line 38</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="scrapePageRecipe"><span class="type-signature">(async) </span>scrapePageRecipe<span class="type-signature"></span></h4>




<div class="description">
    Creates a scraper for the user's current page and parsesdata from it.
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line53">line 53</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
            
<h4 class="name" id="updateRecipe"><span class="type-signature"></span>updateRecipe<span class="type-signature"></span></h4>




<div class="description">
    Updates state.recipe to the newRecipe param
</div>







<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line68">line 68</a>
    </li></ul></dd>
    

    

    

    
</dl>






        
    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="renderSaveSuccess"><span class="type-signature"></span>renderSaveSuccess<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Renders save success message if state.saveSuccess is truthy
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line89">line 89</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
            

    

    
    <h4 class="name" id="renderScrapeError"><span class="type-signature"></span>renderScrapeError<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Renders state.scrapeError if there is an error
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_App_index.js.html">components/App/index.js</a>, <a href="components_App_index.js.html#line75">line 75</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="AllRecipesScraper.html">AllRecipesScraper</a></li><li><a href="App.html">App</a></li><li><a href="BaseScraper.html">BaseScraper</a></li><li><a href="Recipe.html">Recipe</a></li><li><a href="RecipeDetails.html">RecipeDetails</a></li><li><a href="SaveRecipeButton.html">SaveRecipeButton</a></li><li><a href="ScraperFactory.html">ScraperFactory</a></li></ul><h3><a href="global.html">Global</a></h3>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Tue May 29 2018 10:40:09 GMT-0500 (Central Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>