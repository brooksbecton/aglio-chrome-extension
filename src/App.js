import React, { Component } from "react";

import SaveRecipeButton from "./SaveRecipeButton";

/** Wrapper for application */
class App extends Component {
  render() {
    return (
      <div>
        <SaveRecipeButton />
      </div>
    );
  }
}

export default App;
