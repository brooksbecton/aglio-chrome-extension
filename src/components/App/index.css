.button {
  margin: 10px;
}

.mainContainer {
  padding: 3%;
  width: 600px;
}

.center {
  margin: auto;
  width: 100%;
}
