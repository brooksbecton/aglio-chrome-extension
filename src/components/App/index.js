import React, { Component } from "react";
import Button from "react-bootstrap/lib/Button";

import "./index.css";
import RecipeDetails from "./../RecipeDetails";
import ScraperFactory from "./../../lib/scrapers/ScraperFactory";
import Recipe from "./../../lib/models/Recipe";

/** Wrapper for application */
class App extends Component {
  /** Creates an instance of ScraperFactory and attaches it to component */
  constructor() {
    super();
    this.state = {
      recipe: {},
      recipeLoaded: false,
      saveSuccess: null,
      saveError: null,
      scrapeError: null
    };
    this.recipeModel = new Recipe();
    this.scraperFactory = new ScraperFactory();
  }

  componentDidMount() {
    /** Scrapes the page when the extension is opened */
    this.scrapePageRecipe();
  }

  closeExtension() {
    window.close();
  }

  /**
   * Save the component's state to the database
   * @async
   */
  saveRecipe = async () => {
    const { recipe } = this.state;
    try {
      await this.recipeModel.insert(recipe);
      this.setState({ saveSuccess: true });
    } catch (err) {
      this.setState({ saveSuccess: false, saveError: err });
    }
  };

  /**
   * Creates a scraper for the user's current page and parses
   * data from it.
   * @async
   */
  scrapePageRecipe = async () => {
    let scraper;
    try {
      scraper = await this.scraperFactory.create();
      const recipe = await scraper.run();
      this.setState({ recipe, recipeLoaded: true });
    } catch (err) {
      this.setState({ scrapeError: err.toString() });
    }
  };

  /**
   * Updates state.recipe to the newRecipe param
   * @param {Object} - recipe
   */
  updateRecipe = newRecipe => {
    this.setState({ recipe: newRecipe });
  };

  /**
   * Renders state.scrapeError if there is an error
   */
  renderScrapeError() {
    if (this.state.scrapeError) {
      return (
        <div className="center">
          <h1>Error</h1>
          <p>{this.state.scrapeError}</p>
        </div>
      );
    }
  }

  /**
   * Renders save success message if state.saveSuccess is truthy
   */
  renderSaveSuccess() {
    if (this.state.saveSuccess) {
      return (
        <div className="center">
          <h1>Save Success</h1>
          <Button bsStyle="primary" onClick={() => this.closeExtension()}>
            Close
          </Button>
        </div>
      );
    }
  }

  render() {
    return (
      <div className="mainContainer">
        {this.renderScrapeError()}
        {this.renderSaveSuccess()}

        {!this.state.saveSuccess &&
          this.state.recipeLoaded && (
            <div>
              <Button
                bsStyle="success"
                className={["button", "saveButton"]}
                onClick={() => this.saveRecipe()}
              >
                Save
              </Button>
              <br />
              <RecipeDetails
                recipe={this.state.recipe}
                updateRecipe={this.updateRecipe}
              />
            </div>
          )}
        <div className="recipeHtml" />
      </div>
    );
  }
}

export default App;
