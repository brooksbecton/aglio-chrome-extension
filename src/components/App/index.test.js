import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import App from "./index";

Enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("matches recipe loaded snapshot", () => {
    const tree = shallow(<App />);

    tree.setState({
      saveSuccess: false,
      recipeLoaded: true,
      recipe: {
        recipe: ["eat", "sleep", "code"],
        ingredients: ["JavaScript", "vscode", "jest"],
        meta: {
          cookTime: "24h 60m"
        }
      },
      scrapeError: null
    });

    expect(tree).toMatchSnapshot();
  });

  it("matches save success snapshot", () => {
    const tree = shallow(<App />);

    tree.setState({
      saveSuccess: true
    });

    expect(tree).toMatchSnapshot();
  });

  it("matches 'No Scraper' snapshot", () => {
    const tree = shallow(<App />);

    tree.setState({
      saveSuccess: null,
      recipeLoaded: null,
      scrapeError: "Jest offers no recipes!"
    });

    expect(tree).toMatchSnapshot();
  });
});
