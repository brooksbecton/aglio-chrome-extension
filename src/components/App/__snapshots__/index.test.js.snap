// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches 'No Scraper' snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="center"
>
          <h1>
                    Error
          </h1>
          <p>
                    Jest offers no recipes!
          </p>
</div>,
        undefined,
        null,
        <div
          className="recipeHtml"
/>,
      ],
      "className": "mainContainer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Error
</h1>,
            <p>
              Jest offers no recipes!
</p>,
          ],
          "className": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Error",
            },
            "ref": null,
            "rendered": "Error",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Jest offers no recipes!",
            },
            "ref": null,
            "rendered": "Jest offers no recipes!",
            "type": "p",
          },
        ],
        "type": "div",
      },
      undefined,
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "recipeHtml",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="center"
>
            <h1>
                        Error
            </h1>
            <p>
                        Jest offers no recipes!
            </p>
</div>,
          undefined,
          null,
          <div
            className="recipeHtml"
/>,
        ],
        "className": "mainContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Error
</h1>,
              <p>
                Jest offers no recipes!
</p>,
            ],
            "className": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Error",
              },
              "ref": null,
              "rendered": "Error",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Jest offers no recipes!",
              },
              "ref": null,
              "rendered": "Jest offers no recipes!",
              "type": "p",
            },
          ],
          "type": "div",
        },
        undefined,
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "recipeHtml",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App matches recipe loaded snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        undefined,
        <div>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="success"
                    className={
                              Array [
                                        "button",
                                        "saveButton",
                                      ]
                    }
                    disabled={false}
                    onClick={[Function]}
          >
                    Save
          </Button>
          <br />
          <RecipeDetails
                    recipe={
                              Object {
                                        "ingredients": Array [
                                          "JavaScript",
                                          "vscode",
                                          "jest",
                                        ],
                                        "meta": Object {
                                          "cookTime": "24h 60m",
                                        },
                                        "recipe": Array [
                                          "eat",
                                          "sleep",
                                          "code",
                                        ],
                                      }
                    }
                    updateRecipe={[Function]}
          />
</div>,
        <div
          className="recipeHtml"
/>,
      ],
      "className": "mainContainer",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="success"
              className={
                            Array [
                                          "button",
                                          "saveButton",
                                        ]
              }
              disabled={false}
              onClick={[Function]}
>
              Save
</Button>,
            <br />,
            <RecipeDetails
              recipe={
                            Object {
                                          "ingredients": Array [
                                            "JavaScript",
                                            "vscode",
                                            "jest",
                                          ],
                                          "meta": Object {
                                            "cookTime": "24h 60m",
                                          },
                                          "recipe": Array [
                                            "eat",
                                            "sleep",
                                            "code",
                                          ],
                                        }
              }
              updateRecipe={[Function]}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "success",
              "children": "Save",
              "className": Array [
                "button",
                "saveButton",
              ],
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Save",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "recipe": Object {
                "ingredients": Array [
                  "JavaScript",
                  "vscode",
                  "jest",
                ],
                "meta": Object {
                  "cookTime": "24h 60m",
                },
                "recipe": Array [
                  "eat",
                  "sleep",
                  "code",
                ],
              },
              "updateRecipe": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "recipeHtml",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          undefined,
          <div>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="success"
                        className={
                                    Array [
                                                "button",
                                                "saveButton",
                                              ]
                        }
                        disabled={false}
                        onClick={[Function]}
            >
                        Save
            </Button>
            <br />
            <RecipeDetails
                        recipe={
                                    Object {
                                                "ingredients": Array [
                                                  "JavaScript",
                                                  "vscode",
                                                  "jest",
                                                ],
                                                "meta": Object {
                                                  "cookTime": "24h 60m",
                                                },
                                                "recipe": Array [
                                                  "eat",
                                                  "sleep",
                                                  "code",
                                                ],
                                              }
                        }
                        updateRecipe={[Function]}
            />
</div>,
          <div
            className="recipeHtml"
/>,
        ],
        "className": "mainContainer",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="success"
                className={
                                Array [
                                                "button",
                                                "saveButton",
                                              ]
                }
                disabled={false}
                onClick={[Function]}
>
                Save
</Button>,
              <br />,
              <RecipeDetails
                recipe={
                                Object {
                                                "ingredients": Array [
                                                  "JavaScript",
                                                  "vscode",
                                                  "jest",
                                                ],
                                                "meta": Object {
                                                  "cookTime": "24h 60m",
                                                },
                                                "recipe": Array [
                                                  "eat",
                                                  "sleep",
                                                  "code",
                                                ],
                                              }
                }
                updateRecipe={[Function]}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "success",
                "children": "Save",
                "className": Array [
                  "button",
                  "saveButton",
                ],
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Save",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "recipe": Object {
                  "ingredients": Array [
                    "JavaScript",
                    "vscode",
                    "jest",
                  ],
                  "meta": Object {
                    "cookTime": "24h 60m",
                  },
                  "recipe": Array [
                    "eat",
                    "sleep",
                    "code",
                  ],
                },
                "updateRecipe": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "recipeHtml",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`App matches save success snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          className="center"
>
          <h1>
                    Save Success
          </h1>
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    disabled={false}
                    onClick={[Function]}
          >
                    Close
          </Button>
</div>,
        false,
        <div
          className="recipeHtml"
/>,
      ],
      "className": "mainContainer",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Save Success
</h1>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="primary"
              disabled={false}
              onClick={[Function]}
>
              Close
</Button>,
          ],
          "className": "center",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Save Success",
            },
            "ref": null,
            "rendered": "Save Success",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "primary",
              "children": "Close",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Close",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "recipeHtml",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            className="center"
>
            <h1>
                        Save Success
            </h1>
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="primary"
                        disabled={false}
                        onClick={[Function]}
            >
                        Close
            </Button>
</div>,
          false,
          <div
            className="recipeHtml"
/>,
        ],
        "className": "mainContainer",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Save Success
</h1>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                disabled={false}
                onClick={[Function]}
>
                Close
</Button>,
            ],
            "className": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Save Success",
              },
              "ref": null,
              "rendered": "Save Success",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "primary",
                "children": "Close",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Close",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "recipeHtml",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
